### PR TITLE
Fix equals columns check bug

### LINKED ISSUE #5277: Query of `tpcds` execute error.
### Steps to reproduce the behavior (Required)
1. CREATE TABLE '...'
tpcds sql.
2. INSERT INTO '....'
3. SELECT  '....'
query24.sql
query49.sql
### Expected behavior (Required)
execute correctly.
### Real behavior (Required)
ERROR 1064 (HY000) at line 2: Index: 1, Size: 1
### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - 6d14afa

### PR DESCRIPTION
**Pull request by Seaven**
## What type of PR is this：
- [x] bug
- [ ] feature
- [ ] enhancement
- [ ] others

## Which issues of this PR fixes ：

Fixes #5277

## Problem Summary(Required) ：

Check property error

1. e.g:
A join B on a = b, A is local(a, b),  local(a, b) can't satisfy shuffle (a)

2. e.g:
A join B on a1 = b1, a2 = b2, A is local(a1), local(a1) is satisfy shuffle(a1, b1)

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionSpec.java [hunk 1]
```diff
@@ -31,31 +32,42 @@ private boolean isJoinEqColumnsCompatible(HashDistributionSpec requiredSpec) {
...
+                 * outputColumn: A.a
+                 * joinEquivalentColumns: [A.a, B.b, C.c, D.d]
+                 *
+                 * A.a can be mapped to D.d through joinEquivalentColumns
+                 */
+                if (!propertyInfo.isEquivalentJoinOnColumns(requiredShuffleColumn, outputShuffleColumn)) {
+                    return false;
+                }
             }
         }
```
Comment (liuyehcf) [inline fe/fe-core/src/main/java/com/starrocks/sql/optimizer/base/HashDistributionSpec.java:68]: else if block missing return true?

**Review by kangkaisen — DISMISSED** [review]

**Review by liuyehcf — DISMISSED** [review]

**Review by Youngwb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage check]

:heart_eyes: **pass** : 16 / 16 (100.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/DistributionSpec.java|1|1|100.00%|
|:large_blue_circle:|com/starrocks/sql/optimizer/base/HashDistributionSpec.java|15|15|100.00%|

**Comment by Seaven** [comment]
sh starrocks_be_unittest

**Comment by Seaven** [comment]
run starrocks_be_unittest

**Review by Youngwb — APPROVED** [review]

**Review by kangkaisen — APPROVED** [review]

**Pull request merged**